corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -120,3 +170,23 @@ def available_for_domain(cls, domain):
             super(ArcGISFormExpressionRepeater, cls).available_for_domain(domain)
             and ARCGIS_INTEGRATION.enabled(domain)
         )
+
+
+def get_evaluation_context(domain, repeat_record, payload_doc, response):
```

Comment:
nit: Is there any expectation about what should be present in the Evaluation Context?

When i was making [this change](https://github.com/dimagi/commcare-hq/pull/34963/commits/0112c7f27b0336d07c07fc9e5b1a6b1d28b21b71#diff-d3e6fd65dcf035371e2a16212a7e32133aad431f872961e5b5fbc44c258f2001R133) and I had to add domain to the context because it was expected for `RelatedDocExpressionSpec` [here](https://github.com/dimagi/commcare-hq/blob/fb70ab0a59aef910553938067bd373b543ebcabe/corehq/apps/userreports/expressions/specs.py#L611), it felt that the root_doc in EvaluationContext is probably expected to be a form/case/any tracked domain bound model. But may be that isn't being retained anymore and the evaluation context's root_doc is made a data collection instead that is needed.

Is that the right understanding?

Reply:
That's basically correct. The evaluation context depends on the context it's being used in. For UCRs it is the JSON representation of the data model (case, form, user etc). In other contexts the specifics are determined solely by that context.

If there are some global expectations then it could make sense to validate those in `EvaluationContext` (though there should be a way to circumvent that for special cases e.g. tests)